README.md

```diff
@@ -4,57 +4,120 @@ This is a tiny little shim around Cosmos SDK binaries that use the upgrade
+active folder (so `current/bin/$DAEMON_NAME` is the binary)
 
+Note: the `<name>` after `upgrades` is the URI-encoded name of the upgrade as specified in the upgrade module plan.
 
-## Layout of `DAEMON_HOME` Folder
+**Question** where is the state of the blockchain (config, data) stored? Under the update manager, or elsewhere?
 
-* `$DAEMON_HOME/upgrade_manager/genesis` or `$DAEMON_HOME/upgrade_manager/genesis/bin/$DAEMON_NAME` should point to the
-genesis binary
-* `$DAEMON_HOME/upgrade_manager/upgrades/<upgrade-name>` or
-`$DAEMON_HOME/upgrade_manager/upgrades/<upgrade-name>/bin/$DAEMON_NAME` should point to the binary for the upgrade
-named `<upgrade-name>`
-* `$DAEMON_HOME/upgrade_manager/current` is symlink'ed to the current binary
+## Usage
 
+Basic Usage:
+
+* The admin is responsible for installing the `upgrade_manager` and setting it as a eg. systemd service to auto-restart
+* The admin is responsible for installing the `genesis` folder manually
+* The admin is responsible for setting the `current` link to point to genesis
+* The admin is (generally) responsible for installing the `upgrades/<name>` folders manually
+* The upgrade manager handles switching over the binaries at the correct points, so the admin can prepare days in advance and relax at upgrade time
+
+Note that chains that wish to support upgrades may package up a genesis upgrade manager tar file with this info, just as they
+prepare the genesis binary tar file. In fact, they may offer a tar file will all upgrades up to current point for easy download
+for those who wish to sync a fullnode from start.
+
+The `DAEMON` specific code, like the tendermint config, the application db, syncing blocks, etc is done as normal.
+The same eg. `GAIA_HOME` directives and command-line flags work, just the binary name is different.
 
 ## Upgradeable Binary Specification
 
-This upgrade manager works with any binary that follows the following specification:
+In the basic version, the upgrade_manager will read the stdout log messages
+to determine when an upgrade is needed. We are considering more complex solutions
+via signaling of some sort, but starting with the simple design:
 
-* when an upgrade is needed the binary will emit a message that matches this
+* when an upgrade is needed the binary will print a line that matches this
 regular expression: `UPGRADE "(.*)" NEEDED at height (\d+):(.*)`.
 * the second match in the above regular expression can be a JSON object with
 a `binaries` key as described above
 
-Binaries will be started as sub-processes and killed when an upgrade is needed
+The name (first regexp) will be used to select the new binary to run. If it is present,
+the current subprocess will be killed, `current` will be upgraded to the new directory, 
+and the new binary will be launched.
+
+**Question** should we just kill the upgrade manager after it does the updates?
```

Comment:
I think it's fine to just kill the sub-process and start a new one. Any reason that would be a bad idea?

Reply:
Generally long-running processes consume resources.. expecially if they have go routines reading megabytes of logs... Since we do a restart here anyway, it is a nice place to refresh the manager as well.

Not required, but a suggestion. We should clarify it either way

Reply:
If we do this approach, this would require `systemd` to be used to restart the upgrade manager, right?